app/bundles/InstallBundle/Configurator/Step/EmailStep.php

```diff
@@ -166,7 +140,9 @@ public function update(StepInterface $data)
         $parameters = [];
 
         foreach ($data as $key => $value) {
-            $parameters[$key] = $value;
+            if (!in_array($key, self::SKIP_PARAMETERS)) {
```

Comment:
We were chatting with @mollux that the email step should be removed from the installation steps. Now that there will be just the SMTP option it may be better to just remove it. It sucks to force users to configure a SMTP transport when they want to use another transport that they install as a plugin once the installation is complete.

@RCheesley WDYT?

Reply:
I think it's an interesting idea, but how will someone configure SMTP if they want to use just that, and not any plugin? If we remove this step, how will they know that they have to do this before email of any kind is going to be sent? That is my main concern here really, it would add confusion for people I feel, especially those new to Mautic.

I would suggest we have an option for will you use SMTP or a third party mailer, and only show SMTP settings if they choose that option, maybe. We can inform them that they will have the option to install other plugins once Mautic is installed?

Reply:
Then we can remove it and we can inform them that they will have the options to setup it once Mautic is installed? Let's simplify it

Reply:
Exactly. It would move from enforcing email configuration in the installation to the documentation. Same as cron setting for example. Mautic is useless without configuring cron but it is not part of the installation itself.

Reply:
I support removing the email step, right now we have a very easy-to-use setup, but with the new mailer things are more complex and DSN string that you should enter could include lots of configurations like this
```ses+api://ACCESS_KEY:SECRET_KEY@default```
```failover(postmark+api://ID@default sendgrid+smtp://KEY@default)"```
```"roundrobin(postmark+api://ID@default sendgrid+smtp://KEY@default)"```

Do we really want to have a visual builder to build these strings? I wrote in this PR a basic converter that works with a basic string but for others, there is no support yet

Reply:
Let's finalize this discussion, to summarize

- There is a decision to remove the EmailStep from the Install Process
- In this PR there is a visual builder to build `DSN String` that configures `symfony/mailer`, this is a very basic builder and supports one transport, for those who want to use advanced configurations like `failover` or `roundrobin` they have to do the strings themselves as this UI build will fail

Reply:
@mabumusa1 I think this is fine providing that we give very clear instructions in the end-user documentation including code samples to help people understand how to configure the more complex parts.

Reply:
The email step was removed in https://github.com/mautic/mautic/pull/12216